src/plugins/visualizations/public/legacy/build_pipeline.ts

```diff
@@ -331,7 +331,20 @@ const buildVisConfig: BuildVisConfigFunction = {
   },
 };
 
-export const buildVislibDimensions = async (vis: any, params: BuildPipelineParams) => {
+export interface VislibDimensions {
+  x: any;
+  y: SchemaConfig[];
+  z: any[] | undefined;
+  width: any[] | undefined;
+  series: any[] | undefined;
+  splitRow: any[] | undefined;
+  splitColumn: any[] | undefined;
```

Comment:
nit - can't these just be optional params?

```suggestion
  z?: any[];
  width?: any[];
  series?: any[];
  splitRow?: any[];
  splitColumn?: any[];
```